lib/mongodb.js

```diff
@@ -1488,15 +1488,26 @@ MongoDB.prototype.count = function count(modelName, where, options, callback) {
     debug('count', modelName, where);
   }
   where = self.buildWhere(modelName, where, options) || {};
-  const method = Object.keys(where).length === 0 ? 'estimatedDocumentCount' : 'countDocuments';
```

Comment:
Why not use 
```ts
options = buildOptions({}, options);
const method = (Object.keys(where).length === 0  && !options.session) ? 'estimatedDocumentCount' : 'countDocuments';
```
To simplify/remove your new condition?

Reply:
execute with options has different signature for estimatedDocumentCount and countDocuments. check the code below.